base64/base64.go

```diff
@@ -1,12 +1,37 @@
 // Package base64 implements base64 encoding as specified by RFC 4648.
 package base64
 
+const alphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/"
+
 // Encode encodes src using base64 encoding.
 func Encode(src []byte) (dst []byte) {
-	return nil
+	//for len(src)%3 != 0 {
+	//src = append(src, 61)
+	//}
+	for a, i := 0, 1; a < len(src); a++ {
```

Comment:
I wouldn't say that this implementation is the most intuitive and easy to understand, but it's a nice idea.

What you implemented is called a state machine: https://en.wikipedia.org/wiki/Finite-state_machine.